community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -893,6 +893,24 @@ public void visitStore( Visitor<CommonAbstractStore,RuntimeException> visitor )
         visitor.visit( this );
     }
 
+    /**
+     * Called from the part of the code that starts the {@link NeoStore} and friends, together with any
+     * existing transaction log, seeing that there are transactions to recover. Now, this shouldn't be
+     * needed because the state of the id generator _should_ reflect this fact, but turns out that,
+     * given HA and the nature of the .id files being like orphans to the rest of the store, we just
+     * can't trust that to be true. If we happen to have id generators open during recovery we delegate
+     * {@link #freeId(long)} calls to {@link IdGenerator#freeId(long)} and since the id generator is most likely
+     * out of date w/ regards to high id, it may very well blow up.
+     */
+    final void recoveryRequired()
```

Comment:
`resetIdGenerator` is quite implementation-specific, why do we need to reflect that in the method name, I don't get that. Also `reset` doesn't say much.